<h2>Soal 1: Membuat Objek DataFrames</h2>

- Buatlah objek dataframes dari objek dictionary, Bebas!

In [2]:
# Membuat dataframes dari dictionary
import pandas as pd

# Directory
dir = {'tipe_int' : [1,2,3,4,5], 'tipe_string' : ['a','b','c','d','e']}
#DataFrame
df = pd.DataFrame(dir)
print("isi direktori adalah: ")
print(dir)
print("")
print("isi data frame adalah:")
print(df)

isi direktori adalah: 
{'tipe_int': [1, 2, 3, 4, 5], 'tipe_string': ['a', 'b', 'c', 'd', 'e']}

isi data frame adalah:
   tipe_int tipe_string
0         1           a
1         2           b
2         3           c
3         4           d
4         5           e


<h2>Soal 2: Membaca data dari file</h2>

Download file bernama [diamonds.csv](https://drive.google.com/uc?export=download&id=1qQiM9utQUThTPq8jk8stPMh_y2zfqYOd), kemudian baca data tersebut dan assign ke dalam suatu variabel kemudian print.

In [4]:
# Membaca data dan memasukannya kedalam suatu variabel
import pandas as pd
df_from_csv = pd.read_csv('diamonds.csv', delimiter = ",")
df_from_csv

Unnamed: 0  carat        cut color clarity  depth  table   price     x  \
0              1   0.23      Ideal     E     SI2   61.5   55.0   326.0  3.95   
1              2   0.21    Premium     E     SI1   59.8   61.0   326.0  3.89   
2              3   0.23       Good     E     VS1   56.9   65.0   327.0  4.05   
3              4   0.29    Premium     I     VS2   62.4   58.0   334.0  4.20   
4              5   0.31       Good     J     SI2   63.3   58.0   335.0  4.34   
...          ...    ...        ...   ...     ...    ...    ...     ...   ...   
6482        6483   1.22  Very Good     J     SI2   62.2   60.0  4058.0  6.77   
6483        6484   1.24       Fair     G     SI2   65.7   57.0  4058.0  6.72   
6484        6485   0.90  Very Good     G     SI1   62.7   59.0  4059.0  6.12   
6485        6486   1.04      Ideal     J     SI1   60.8   57.0  4059.0  6.54   
6486        6487   0.90      Ideal     G     NaN    NaN    NaN     NaN   NaN   

         y     z  
0     3.98  2.43  
1     3.84  2.31  
2     4.07  2.31  
3     4.23  2.63  
4     4.35  2.75  
...    ...   ...  
6482  6.89  4.25  
6483  6.68  4.40  
6484  6.17  3.85  
6485  6.59  3.99  
6486   NaN   NaN  

[6487 rows x 11 columns]

Expected Result:

![alt text](https://drive.google.com/uc?id=1BvgM5mJEVfj0fllrjkBfQwn0QIFkry_t)

<h2>Soal 3: DataFrame vs Series</h2>

- Apa perbedaan Series dengan list dan dictionary
- Apa perbedaan dataframe dan series
- Buatlah suatu data series

Jawab: 
Series bisa dibilang hampir sama seperti list dan berbeda dengan dictionary, karena Series dan list sama-sama berbentuk 1 dimensi hanya saja Series hanya bisa menggunakan pandas. Tidak seperti list dan dictionary bisa langsung menggunakan python tanpa module pandas.

perbedaan antara dataframe dan series adalah pada dataframe isi datanya bersifat 2 dimensi, sedangkan pada series bersifat 1 dimensi

In [5]:
# code here
import pandas as pd
pan_series = pd.Series([1,3,5,7,9], index=['a','b','c','d','e'])
print(pan_series)

a    1
b    3
c    5
d    7
e    9
dtype: int64


<h2>Soal 4: Pemeriksaan Data Sederhana</h2>

Di soal no 2 kamu telah membaca data tentang diamonds. pada soal ini cobalah suatu metode dari pandas untuk mengetahui beberapa data pertama dan beberapa data terakhir. kemudian sebutkan nilai dari:
- baris pertama dari column price
- baris terakhir dari column color

tunjukan cara kalian menemukan nilai tersebut di block di bawah ini.

In [6]:
# code here
price = df_from_csv['price']
price.head(1)

0    326.0
Name: price, dtype: float64

In [7]:
color = df_from_csv['color']
color.tail(1)

6486    G
Name: color, dtype: object

<h2>Soal 5: Deskriptif Statistik</h2>

Jelaskan apa itu deskriptis statistik!

Jawab: metode dari pengorganisasian, penjumlahan, dan penyajian data dalam sebuah cara yang nyaman dan informatif, termasuk teknik grafik, dan teknik penghitungan. Statistik deskriptif dapat mendeskripsikan data yang sedang dianalisis, tetapi tidak boleh menarik kesimpulan apapun dari data.

<h2>Soal 6: Practice Deskriptif Statistik</h2>

- Gunakan suatu metode dari pandas untuk mengetahui deskriptif statistik dari suatu data
- Berapa nilai rata-rata dari column price
- Berapa nilai standar deviasi dari column depth
- Berapa nilai maximum dari column carat

In [12]:
#code here
df_from_csv.describe()

Unnamed: 0        carat        depth        table        price  \
count  6487.00000  6487.000000  6486.000000  6486.000000  6486.000000   
mean   3244.00000     0.792539    61.859297    57.728970  3073.217700   
std    1872.77993     0.216498     1.670383     2.373592   946.945215   
min       1.00000     0.200000    43.000000    49.000000   326.000000   
25%    1622.50000     0.710000    61.000000    56.000000  2915.000000   
50%    3244.00000     0.800000    61.900000    57.000000  3278.500000   
75%    4865.50000     0.910000    62.700000    59.000000  3674.000000   
max    6487.00000     1.520000    71.600000    70.000000  4059.000000   

                 x            y            z  
count  6486.000000  6486.000000  6486.000000  
mean      5.862063     5.862949     3.624252  
std       0.621652     0.610162     0.399987  
min       3.790000     3.750000     0.000000  
25%       5.720000     5.750000     3.530000  
50%       5.960000     5.960000     3.680000  
75%       6.230000     6.230000     3.870000  
max       7.560000     7.420000     4.800000

In [11]:
print('nilai rata - rata dari column price adalah')
price.mean()

nilai rata - rata dari column price adalah


3073.2176996608077

In [9]:
print('nilai standar deviasi dari column depth adalah')
depth = df_from_csv['depth']
depth.std()

nilai standar deviasi dari column depth adalah


1.6703830554400516

In [10]:
print('nilai max dari column carat adalah')
carat = df_from_csv['carat']
carat.max()

nilai max dari column carat adalah


1.52

Jawab:
1. nilai rata - rata dari column price adalah 3073.2176996608077
2. nilai standar deviasi dari column depth adalah 1.6703830554400516
3. nilai max dari column carat adalah 1.52